esp-hal/src/timer/timg.rs

```diff
@@ -1040,6 +1091,169 @@ where
     }
 }
 
+// Async functionality of the timer groups.
+mod asynch {
+    use core::{
+        pin::Pin,
+        task::{Context, Poll},
+    };
+
+    use embassy_sync::waitqueue::AtomicWaker;
+    use procmacros::handler;
+
+    use super::*;
+
+    cfg_if::cfg_if! {
+        if #[cfg(all(timg1, timg_timer1))] {
+            const NUM_WAKERS: usize = 4;
+        } else if #[cfg(timg1)] {
+            const NUM_WAKERS: usize = 2;
+        } else {
+            const NUM_WAKERS: usize = 1;
+        }
+    }
+
+    #[allow(clippy::declare_interior_mutable_const)]
```

Comment:
is this still needed?

Reply:
_Probably_ not but we have it on multiple places so I can address it in a separate PR.

Reply:
In the other places that IS NEEDED but your previous change from removing `const INIT: AtomicWaker = AtomicWaker::new();` made this obsolete - so it's a thing of this PR

Reply:
Oops, I looked into not up-to-date revision and those I saw were addressed in https://github.com/esp-rs/esp-hal/pull/2200. Sorry. 
Updated, thanks!